add placeholders

aquibyatoo opened the pull request:
### Status :
 In Development 

### Description :
Add placeholders for loading state for app. There are Box, Line, Image Card, Card and Paragraph placeholders.

##
### Related Issues :
#169

## Todos
- [ ] Tests
- [ ] Documentation

Commit:
add placeholders

Commit:
add placeholders

Commit:
add lineWidth in text placeholder

Commit:
add lineWidth in text placeholder

rishichawda reviewed (changes requested):
@aquibyatoo Can you add documentation so that it is possible for me to test it out quickly? You can refer other component directories to get an idea on how to document it.

Commit:
add placeholders tests

Commit:
add placeholders tests

Commit:
add readMe.md files

Commit:
add readMe.md files

rishichawda commented on lib/placeholders/box/readMe.md:
```diff
@@ -0,0 +1,31 @@
+
+| Name     | Type     | Default   | Description                          |
+| :------- | :------- | :-------- | :----------------------------------- |
+| `square` | `circle` | &nbsp;    | Shapes of the box default is circle. |
```
I can't see `square` prop used anywhere in the component. Is this by mistake?

aquibyatoo replied:
07e1bd894e713bbe71d96aaf554badda90d852db

rishichawda commented on .gitignore:
```diff
@@ -6,4 +6,6 @@ coverage
-/components
\ No newline at end of file
+/components
+/.prettierrc.json
```
why was prettier used? We already have eslint set up and an eslintrc on the root.

rishichawda replied:
Please make sure code is formatted using eslint and the config that is already present.

aquibyatoo replied:
321bd4ced83925829ae7a889f3d16964b6fa0c12

Commit:
fix readMe.md of box placeholder

Commit:
fix readMe.md of box placeholder

rishichawda commented on lib/globals/_theme.scss:
```diff
@@ -55,3 +56,14 @@ $dark-hot-pink-20: rgba(225, 0, 80, 0.2);
+$shimmer-color-light: #f6f7f8;
+$shimmer-color-dark: #edeef1;
+
+// *********** BOX SHADOWS ****************
```
Comments for single lines can be provided at the end of the line. Comments like this increase the code length (Also try avoiding unnecessary line gaps). Moreover, The names can be descriptive to be understandable at first glance, so probably we can remove comments altogether and just use names that are descriptive enough.

aquibyatoo replied:
321bd4ced83925829ae7a889f3d16964b6fa0c12

rishichawda commented on lib/globals/animations.module.scss:
```diff
@@ -1,41 +1,75 @@
-        100% {
-            width: 100%;
-        }
+  -o-animation-name: animateFocus;
```
Looks like this was due to prettier. Can you try removing prettier and lint files with eslint and see if it reverts these changes?

aquibyatoo replied:
321bd4ced83925829ae7a889f3d16964b6fa0c12

aquibyatoo replied:
0e0da4dd32e77d0afd2b322cf30853314dd207e8

Commit:
fix formatting

Commit:
fix formatting

KRRISH96 commented on lib/placeholders/box/readMe.md:
```diff
@@ -0,0 +1,30 @@
+## Navbar
```
I guess This is added while taking a markdown sample of other components.
Please replace with a relevant title for this component (ex: Box Placeholder).

aquibyatoo replied:
977c4770419b2a5b4289c7e62a8812e8b16cf095

KRRISH96 commented on lib/placeholders/box/readMe.md:
```diff
@@ -0,0 +1,30 @@
+  };
+
+  return <div {...boxPlaceholderProps} />;
+}
```
The Usage Section should be an example of how the component can be used.
```jsx
class Demo extends React.Component {
    state = {
       loading: false,
    }
    render() {
       const {loading} = this.state;
       return (
            <div>
                { loading ? <BoxPlaceholder /> : <div>User Content</div>
             </div>
        )
    }
}
```

Please have look at other components documentation to get an Idea.
Thanks.

aquibyatoo replied:
1309bc5f877031f4e7719566dc665b2940b86281

Commit:
fix readMe.md files

Commit:
fix readMe.md files

KRRISH96 commented on lib/placeholders/card/index.js:
```diff
@@ -0,0 +1,37 @@
+  imageShape: 'circle',
+};
+
+export default themr('CARPLACEHOLDER', defaultTheme)(CardPlaceholder);
```
Looks to be a Typo, `CARDPLACEHOLDER`. 
This doesn't affect the component. Still good to avoid typos.

aquibyatoo replied:
977c4770419b2a5b4289c7e62a8812e8b16cf095

KRRISH96 commented on lib/placeholders/card/readMe.md:
```diff
@@ -0,0 +1,35 @@
+## Navbar
```
Please fix the title.

aquibyatoo replied:
977c4770419b2a5b4289c7e62a8812e8b16cf095

KRRISH96 commented on lib/placeholders/card/readMe.md:
```diff
@@ -0,0 +1,35 @@
+    <LinePlaceholder lineWidth="75" />
+    <LinePlaceholder lineWidth="50" />
+  </div>
+);
```
As mentioned earlier, Please Update Usage here also.

You can have the above usage too, to show how the card placeholder can be constructed using `Box` and `LinePlaceholders` we have.


```jsx
class Demo extends React.Component {
    state = {
       loading: false,
    }
    render() {
       const {loading} = this.state;
       return (
            <div>
                { loading ? <CardPlaceholder /> : <div>User Card</div>
             </div>
        )
    }
}```

aquibyatoo replied:
1309bc5f877031f4e7719566dc665b2940b86281

KRRISH96 commented on lib/placeholders/imageCard/readMe.md:
```diff
@@ -0,0 +1,38 @@
+## Navbar
```
Please Fix title here and in all relevant MD files.

aquibyatoo replied:
977c4770419b2a5b4289c7e62a8812e8b16cf095

KRRISH96 commented on lib/placeholders/imageCard/readMe.md:
```diff
@@ -0,0 +1,38 @@
+      ))}
+    </div>
+  );
+};
```
Similarly, Please Update usage section as mentioned earlier.
Thanks.

aquibyatoo replied:
1309bc5f877031f4e7719566dc665b2940b86281

KRRISH96 reviewed (changes requested):
Please make changes as mentioned in Comments. Thanks.

KRRISH96 commented on lib/placeholders/paragraph/tests/accessibility.test.js:
```diff
@@ -0,0 +1,52 @@
+    expect(wrappedComponent.contains(<div />)).to.equal(false);
+  });
+
+  it('should render the `Lard Placeholder` containing minimum 1 `div` element', () => {
```
Please Fix Typo, `Card Placeholder`.

aquibyatoo replied:
1309bc5f877031f4e7719566dc665b2940b86281

KRRISH96 commented on lib/placeholders/line/tests/accessibility.test.js:
```diff
@@ -0,0 +1,52 @@
+    expect(wrappedComponent.contains(<div />)).to.equal(false);
+  });
+
+  it('should render the `Lard Placeholder` containing minimum 1 `div` element', () => {
```
Please fix Typo, `Card Placeholder`

aquibyatoo replied:
1309bc5f877031f4e7719566dc665b2940b86281

Commit:
fix usage in MD files

Commit:
fix usage in MD files

Commit:
change shape prop in box place holder from defult to square

Commit:
change shape prop in box place holder from defult to square

Commit:
fix tests in box placeholder

Commit:
fix tests in box placeholder

rishichawda reviewed (changes requested):
Regarding the functionalities and current configuration of the components : 

1. I can see there are two components with nearly same function : `ImageCard` and `CardPlaceholder`. I couldn't spot any functional difference between them and they look almost identical to each other too. 

2. Why do we have two exports : `ParagraphPlaceHolder` and `LinePlaceholder` - can it be done in a single component? 

3. `ParagraphPlaceHolder` seems to have box shadow. User might not want it in most of cases probably?

4. The "shimmer" effect looks like it is not completely carried out till the other end of the placeholders.

5. In `CardPlaceholder` the image looks like it has shimmer faster than the lines ( probably because lines are longer than the width of the card ). Can this be tweaked so it looks balanced?

6. The line widths in `ParagraphPlaceHolder` have a fixed value which increments as the lines proceed. Can this be randomised so that it looks more natural?

7. In `CardPlaceholder` component, the line widths are hard-coded. Can this be changed too? Can we also use the existing `Card` component instead of having a separate wrapper again? Since it has other options too like changing shadows, etc. Moreover, it would be then an easier content management from user perspective. Also, we won't have to ship a component like this altogether ( Can be put as a demo using `Card` and `LinePlaceholder` component.

rishichawda committed:
build(dev): greatly improves rebuild times (#338)

incremental builds took a longer time that greatly effected developer experience, this commit improves the rebuild time and generates outputs almost instantaneously.

Commit:
add existing card component in card placeholder and update Md files accordingly

Commit:
merge changes

Commit:
fix shimmer speed and length

Commit:
fix tests

rishichawda reviewed (commented):
> Regarding the functionalities and current configuration of the components :
> 
> I can see there are two components with nearly same function : ImageCard and CardPlaceholder. I couldn't spot any functional difference between them and they look almost identical to each other too.
> 
> Why do we have two exports : ParagraphPlaceHolder and LinePlaceholder - can it be done in a single component?
> 
> ParagraphPlaceHolder seems to have box shadow. User might not want it in most of cases probably?
> 
> The "shimmer" effect looks like it is not completely carried out till the other end of the placeholders.
> 
> In CardPlaceholder the image looks like it has shimmer faster than the lines ( probably because lines are longer than the width of the card ). Can this be tweaked so it looks balanced?
> 
> The line widths in ParagraphPlaceHolder have a fixed value which increments as the lines proceed. Can this be randomised so that it looks more natural?
> 
> In CardPlaceholder component, the line widths are hard-coded. Can this be changed too? Can we also use the existing Card component instead of having a separate wrapper again? Since it has other options too like changing shadows, etc. Moreover, it would be then an easier content management from user perspective. Also, we won't have to ship a component like this altogether ( Can be put as a demo using Card and LinePlaceholder component.

@aquibyatoo Are all the changes present in this branch?
